src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -0,0 +1,2018 @@
+{
+    internal virtual partial void M1();
+}";
+            var comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (4,35): error CS8793: Partial method 'C.M1()' must have an implementation part because it has accessibility modifiers.
+                //     internal virtual partial void M1();
+                Diagnostic(ErrorCode.ERR_PartialMethodWithAccessibilityModsMustHaveImplementation, "M1").WithArguments("C.M1()").WithLocation(4, 35)
+            );
+        }
+
+        [Fact]
+        public void Virtual_Mismatch_01()
+        {
+            const string text1 = @"
+partial class C
+{
+    internal virtual partial void M1();
+    internal partial void M1() { }
+}";
+            var comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (5,27): error CS8798: Both partial method declarations must have equal combinations of 'virtual', 'override', 'sealed', or 'new' modifiers.
+                //     internal partial void M1() { }
+                Diagnostic(ErrorCode.ERR_PartialMethodExtendedModDifference, "M1").WithLocation(5, 27)
+            );
+        }
+
+        [Fact]
+        public void Virtual_Mismatch_02()
+        {
+            const string text1 = @"
+partial class C
+{
+    internal partial void M1();
+    internal virtual partial void M1() { }
+}";
+            var comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (5,35): error CS8798: Both partial method declarations must have equal combinations of 'virtual', 'override', 'sealed', or 'new' modifiers.
+                //     internal virtual partial void M1() { }
+                Diagnostic(ErrorCode.ERR_PartialMethodExtendedModDifference, "M1").WithLocation(5, 35)
+            );
+        }
+
+        [Fact]
+        public void Virtual_Semantics()
+        {
+            const string text1 = @"
+using System;
+
+public partial class C
+{
+    internal virtual partial int M1();
+    internal virtual partial int M1() => 1;
+}
+
+public class D : C
+{
+    internal override int M1() => 2;
+
+    static void UseC(C c)
+    {
+        Console.Write(c.M1());
+    }
+
+    public static void Main()
+    {
+        UseC(new C());
+        UseC(new D());
+    }
+}";
+            CompileAndVerify(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods, expectedOutput: "12").VerifyDiagnostics();
+        }
+
+        [Fact]
+        public void Virtual_AllowNull_01()
+        {
+            const string text1 = @"
+#nullable enable
+using System.Diagnostics.CodeAnalysis;
+
+partial class C
+{
+    internal virtual partial void M1([AllowNull] string s1);
+    internal virtual partial void M1(string s1) { }
+}
+class D : C
+{
+    internal override void M1(string s1) // 1
```

Comment:
What about tests in the other direction: where the `override` is the `partial` member?